src/Core/Repositories/ISecretRepository.cs

```diff
@@ -8,6 +8,7 @@ public interface ISecretRepository
     Task<IEnumerable<Secret>> GetManyByIds(IEnumerable<Guid> ids);
     Task<Secret> GetByIdAsync(Guid id);
     Task<Secret> CreateAsync(Secret secret);
+    Task<Secret> UpdateAsync(Secret secret);
```

Comment:
We should remove `Task ReplaceAsync(Secret secret);` out of the interface and only provide one update method for consumers.